conf_collectd_plugin.py

```diff
@@ -0,0 +1,375 @@
+            "Collectd was not found at our "
+            "default installation folder. "
+            "If you need help with configuration, please "
+            "contact support@wavefront.com\n")
+        sys.exit(1)
+    utils.print_success()
+
+
+def check_collectd_conf_dir():
+    """
+    Check if managed_config directory exists, if not,
+    create one.
+    """
+    res = utils.check_path_exists(COLLECTD_CONF_DIR)
+    if not res:
+        print "Creating collectd managed config dir"
+        utils.call_command("mkdir "+COLLECTD_CONF_DIR)
+
+
+def check_install_state(plugin):
+    print ("Cannot check install state yet.")
+    return False
+
+
+def write_tcpconns_conf_plugin(open_ports):
+    """
+    TODO:
+    -need to check if tcpconn plugin's dependencies are installed
+    -include RemotePort for outbounds connections
+        i.e. the servers we are connecting to
+    """
+    try:
+        out = open("10-tcpconns.conf", "w")
+    except:
+        sys.stderr.write("Unable to write tcpcons.conf file\n")
+        sys.exit(1)
+
+    out.write('LoadPlugin tcpconns')
+    out.write('<Plugin "tcpconns">\n')
+    out.write('  ListeningPorts false\n')
+    for port in open_ports:
+        out.write('  LocalPort "%d"\n' % port)
+    # no remote port yet
+    out.write('</Plugin>\n')
+    out.close()
+
+
+def include_apache_es_conf(dir):
+    """
+    Assume extendedstatus.conf is unique to wavefront and writing over it.
+    """
+    out = utils.write_file(dir+"/extendedstatus.conf")
+    if out is None:
+        utils.exit_with_message("Unexpected error!")
+
+    out.write(
+        '# ExtendedStatus controls whether Apache will generate '
+        '"full" status\n'
+        '# information (ExtendedStatus On) '
+        'or just basic information (ExtendedStatus\n'
+        '# Off) when the "server-status" handler is called. '
+        'The default is Off.\n'
+        'ExtendedStatus on')
+    out.close()
+
+
+# Apache Installation
+def apache_title():
+    sys.stdout.write(
+        " _____ __________  _____  _________   ___ ______________\n"
+        "   /  _  \\______   \/  _  \ \_   ___ \ /   |   \_   _____/\n"
+        "  /  /_\  \|     ___/  /_\  \/    \  \//    ~    \    __)_ \n"
+        " /    |    \    |  /    |    \     \___\    Y    /        \ \n"
+        " \____|__  /____|  \____|__  /\______  /\___|_  /_______  /\n"
+        "         \/                \/        \/       \/        \/ \n")
+
+
+def install_apache_plugin():
+    install = check_install_state("Apache")
+    print ""
+    if not install:
+        sys.stdout.write(
+            "This script has detected that you have apache installed and "
+            "running.")
+        res = utils.ask(
+            "Would you like to run the apache plugin installer to "
+            "enable collectd to collect data from apache?")
+    else:
+        print "You have previously installed this plugin."
+        res = utils.ask("Would you like to reinstall this plugin", no)
```

Comment:
to be more pythonic you can even call `util.ask("Blah", default="no")`.   Actually, this is a bug as it stands.  `no` needs to be `"no"`